src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -208,6 +210,40 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
             // nextLink is the URL needed to get the next page of records using the same query options
             // with $after base64 encoded for opaqueness
             string path = UriHelper.GetEncodedUrl(_httpContextAccessor.HttpContext!.Request).Split('?')[0];
+
+            RuntimeConfig runtimeConfig = _runtimeConfigProvider.GetConfig();
+            // If the base route is not empty, we need to insert it into the URI before the rest path.
+            string? baseRoute = runtimeConfig.Runtime.BaseRoute;
+            if (!string.IsNullOrEmpty(baseRoute))
+            {
+                _sqlMetadataProvider.TryGetEntityPathFromName(context.EntityName, out string? pathNameForEntity);
+
+                // Path is of the form ....restPath/pathNameForEntity. We want to insert the base route before the restPath,
+                // hence we remove pathNameForEntity and restPath from the path and append them back after appending the base route to the path.
+                // Initialize lastIdxOfPathToBeIncluded by removing the pathNameForEntity from the path.
+                int lastIdxOfPathToBeIncluded = path.Length - pathNameForEntity!.Length - 1;
+
+                // Trim any forward slashes present at the end.
+                while (path[lastIdxOfPathToBeIncluded] == '/')
+                {
+                    lastIdxOfPathToBeIncluded--;
+                }
+
+                lastIdxOfPathToBeIncluded = lastIdxOfPathToBeIncluded - runtimeConfig.Runtime.Rest.Path.Length;
+
+                // Initialize pathWithBaseRoute by removing restPath and pathNameForEntity from the path.
+                // Finally, it will be of the form: .../baseRoute/restPath/pathNameForEntity.
+                StringBuilder pathWithBaseRoute = new(path.Substring(0, lastIdxOfPathToBeIncluded + 1));
+
+                // Append base route.
+                pathWithBaseRoute.Append(baseRoute);
+                // Append rest path.
+                pathWithBaseRoute.Append($"{runtimeConfig.Runtime.Rest.Path}");
+                // Append path name for entity.
+                pathWithBaseRoute.Append($"/{pathNameForEntity}");
+                path = pathWithBaseRoute.ToString();
+            }
```

Comment:
If you are trying to just prefix `/<rest-base-path>/<entityName>` with `/<base-path>`, you can do this an easier/less verbose way by:
1. Accessing `_httpContextAccessor.HttpContext!.Request.Path` --> `/api/myEntity` to build your `pathWithBaseRoute`
2. Reference implementation of UriHelper.GetEncodedUrl() used on line 212. Because you don't shouldn't modify the httpcontext directly with your new path value. (that would need more vetting as an option).

Below you would just substitute out request.Path with your generated `pathWithBaseRoute` and strip off the query string.

```csharp
HttpRequest request = _httpContextAccessor.HttpContext!.Request;
string newpath = BuildAbsolute(request.Scheme, request.Host, request.PathBase, pathWithBaseRoute, request.QueryString).Split('?')[0];
```
![image](https://github.com/Azure/data-api-builder/assets/6414189/681aff2f-ec09-4cc4-b2a7-34e032edd982)

Reply:
Or you may even be able to achieve all of this on line 212, and just provide a value for request.pathBase argument with the config set value.

Reply:
Woww, that simplifies things a lot. Thanks for suggesting Sean!

Reply:
Incorporated this!